draft-ietf-lake-edhoc.md

```diff
@@ -338,21 +365,89 @@ If the transport provides a mechanism for correlating messages, some of the conn
+
+When the credential is a COSE_Key, CRED_x is a CBOR map only containing specific fields from the COSE_Key:
+
+* For COSE_Keys of type OKP the CBOR map SHALL only include the parameters 1 (kty), -1 (crv), and -2 (x-coordinate).
+* For COSE_Keys of type EC2 the CBOR map SHALL only include the parameters 1 (kty), -1 (crv), -2 (x-coordinate), and -3 (y-coordinate).
+
+If the parties have agreed on an identity besides the public key, the identity is included in the CBOR map with the label "subject name", otherwise the subject name is the empty text string.
+The parameters SHALL be encoded in decreasing order with int labels first and text string labels last.
+An example of CRED_x when the RPK contains an X25519 static Diffie-Hellman key and the parties have agreed on an EUI-64 identity is shown below:
+
+~~~~~~~~~~~
+CRED_x = {
+  1:  1,
+ -1:  4,
+ -2:  h'b1a3e89460e88d3a8d54211dc95f0b90
+        3ff205eb71912d6db8f4af980d2db83a',
+ "subject name" : "42-50-31-FF-EF-37-32-39"
+}
+~~~~~~~~~~~
+
+## Encoding of Public Authentication Key Identifiers {#id_cred}
+
+ID_CRED_I and ID_CRED_R are identifiers of the public authentication keys of the Initiator and the Responder, respectively. 
+ID_CRED_I and ID_CRED_R do not have any cryptographic purpose in EDHOC.
+
+* ID_CRED_R is intended to facilitate for the Initiator to retrieve the Responder's public authentication key.
+
+* ID_CRED_I is intended to facilitate for the Responder to retrieve the Initiator's public authentication key.
+
+The identifiers ID_CRED_I and ID_CRED_R are COSE header_maps, i.e. CBOR maps containing COSE Common Header Parameters, see Section 3.1 of {{RFC8152}}).
+In the following we give some examples of COSE header_maps.
+
+Raw public keys are most optimally stored as COSE_Key objects and identified with a 'kid' parameter:
+
+* ID_CRED_x = { 4 : kid_x }, where kid_x : bstr, for x = I or R.
+
+Public key certificates can be identified in different ways. Header parameters for identifying X.509 certificates are defined in {{I-D.ietf-cose-x509}}, for example:
+
+* by a hash value with the 'x5t' parameter;
+
+   * ID_CRED_x = { 34 : COSE_CertHash }, for x = I or R,
+
+* by a URL with the 'x5u' parameter;
+
+   * ID_CRED_x = { 35 : uri }, for x = I or R,
+
+
+ID_CRED_x MAY contain the actual credential used for authentication, CRED_x.
+It is RECOMMENDED that they uniquely identify the public authentication key as the recipient may otherwise have to try several keys.
+ID_CRED_I and ID_CRED_R are transported in the ciphertext, see {{m3}} and {{m2}}.
+
+When ID_CRED_x does not contain the actual credential it may be very short.
+One byte credential identifiers are realistic in many scenarios as most constrained devices only have a few keys. In cases where a node only has one key, the identifier may even be the empty byte string.
+
+
+## Identities {#auth-key-id}
```

Comment:
This feels somewhat out of place here... I don't have a good proposal on where to put this text, but as of now it does not read to me as a "protocol element", but either as considerations, or maybe should be under the "Authentication Credentials" as a subsection?